client/src/main/java/org/apache/uniffle/client/impl/ShuffleWriteClientImpl.java

```diff
@@ -922,7 +922,7 @@ public void sendAppHeartbeat(String appId, long timeoutMs) {
           }
           return null;
         },
-        timeoutMs,
+        timeoutMs * allShuffleServers.size() / heartBeatThreadNum + 100,
```

Comment:
The 100 looks a bit arbitrary. In the worst case scenario where all requests timeout those extra 100ms won't give any extra benefit.

Reply:
Please make sure the individual request itself respects the timeout, like in https://github.com/apache/incubator-uniffle/pull/1738/files#diff-4772b95a6a13de40473a9a99cef322eec20df14a4c19182eff997272427154dbR367

Reply:
Maybe you don't need add 100.